VideoLocker/src/main/java/org/edx/mobile/services/ServiceManager.java

```diff
@@ -72,22 +74,25 @@ public HttpRequestEndPoint getEndPointCourseStructure(final String courseId){
         return new HttpRequestEndPoint() {
             public String getUrl() {
                 try {
+                    PrefManager pref = new PrefManager(MainApplication.instance(), PrefManager.Pref.LOGIN);
+                    String username = URLEncoder.encode(pref.getCurrentUserProfile().username, "UTF-8");
                     String block_count = URLEncoder.encode("video", "UTF-8");
-                    String block_fields = URLEncoder.encode("graded,format,multi_device", "UTF-8");
-                    String block_json = URLEncoder.encode("{\"video\":{\"profiles\":[\"mobile_high\",\"mobile_low\"]}}", "UTF-8");
+                    String block_fields = URLEncoder.encode("graded,format,student_view_multi_device", "UTF-8");
+                    String block_json = URLEncoder.encode("video", "UTF-8");
+                    String cId = URLEncoder.encode(courseId, "UTF-8");
```

Comment:
I think these variable names should be changed to be the same as the field names now.